### PR TITLE
[tlog] Adjust app according to the new pi API specs

### PR DESCRIPTION
**Pull request by amassarwi**
This diff adjusts the API routes and their payloads according to the new API 
implemented in https://github.com/decred/politeia/pull/1180

### Main changes:
- Adjust API according to new specs.
- Add separate routes for vetted & unvetted proposal (for details & edit views) as the state is needed to communicate with BE.
- Make it possible to censor public not voting proposals
- Add comments to unvetted propsoals
- Rename UnvettedList to AdminList as it now includes `Censored Vetted` proposal in addition to old `Censored Unvetted` & `Unvetted` proposals. 

*Note:* Currently user's proposals page isn't functioning, as the BE work for it isn't ready yet.

### Dependencies
Needs https://github.com/decred/politeia/pull/1180

### CODE REVIEW

**Review by thi4go — COMMENTED** [review]
Took the PR for a spin and it's looking good mate. Tested all major features with no apparent issues. One thing I did stumble into was an error on the user proposals page  (Account -> Proposals).

`TypeError: Cannot read property 'proposals' of undefined`

**Comment by amassarwi** [comment]
@thi4go thanks for the review, sorry I forgot to mention that the user's proposals route isn't ready yet from BE perspective,
and currently it's not working.

File: src/containers/Proposal/New/New.jsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ const NewProposal = () => {
       <ProposalForm
         disableSubmit={!isPaid || !!identityError}
         onSubmit={onSubmitProposal}
+        isCreate={true}
```
Comment (tiagoalvesdulce) [inline src/containers/Proposal/New/New.jsx:34]: I think we can improve this variable name to `isCreateProposalScreen` or something

File: src/containers/Proposal/User/hooks.js [hunk 2]
```diff
@@ -1,38 +1,38 @@
-import { useEffect, useState, useMemo } from "react";
+import { useEffect, useState /*useMemo*/ } from "react";
```
Comment (tiagoalvesdulce) [inline src/containers/Proposal/User/hooks.js:1]: comment

File: src/containers/Proposal/User/hooks.js [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
 import * as sel from "src/selectors";
 import * as act from "src/actions";
-import { or } from "src/lib/fp";
+//import { or } from "src/lib/fp";
```
Comment (tiagoalvesdulce) [inline src/containers/Proposal/User/hooks.js:4]: comment

File: src/containers/Proposal/User/hooks.js [hunk 4]
```diff
@@ -1,38 +1,38 @@
...
-    const linkedProposal = generalProposals[proposal.linkto];
-    const proposedFor = linkedProposal && linkedProposal.name;
-    return { ...proposal, proposedFor };
-  });
-
-// checks if proposals list from the state machine is incomplete, e.g,
-// there are more user proposals on the redux store than the state machine.
-const isStateMachineMissingProposals = (state, proposals) =>
-  state.proposals && proposals && state.proposals.length !== proposals.length;
+//import useThrowError from "src/hooks/utils/useThrowError";
```
Comment (tiagoalvesdulce) [inline src/containers/Proposal/User/hooks.js:7]: can this go out?

File: src/containers/Proposal/User/hooks.js [hunk 5]
```diff
@@ -67,114 +67,104 @@ export function useDraftProposals() {
...
-
-  useThrowError(error);
-  return {
-    proposals: state.proposals,
-    numOfUserProposals,
-    loading,
-    error,
-    onFetchUserProposals
-  };
+// Curently fetching user's proposals isn't available on the BE side
```
Comment (tiagoalvesdulce) [inline src/containers/Proposal/User/hooks.js:70]: 👍

**Review by tiagoalvesdulce — COMMENTED** [review]
Code is looking good. I left a few suggestions

**Comment by amassarwi** [comment]
to be continued in #2306.

**Pull request closed without merging**